packages/controllers/src/cronjob/CronjobController.test.ts

```diff
@@ -0,0 +1,370 @@
+
+    cronjobController.destroy();
+  });
+
+  it('handles SnapRemoved event', async () => {
+    const rootMessenger = getRootCronjobControllerMessenger();
+    const controllerMessenger =
+      getRestrictedCronjobControllerMessenger(rootMessenger);
+
+    jest.spyOn(controllerMessenger, 'call').mockImplementation((method) => {
+      if (method === 'SnapController:getAll') {
+        return [getTruncatedSnap()];
+      } else if (method === 'PermissionController:getPermissions') {
+        return { [SnapEndowments.Cronjob]: MOCK_CRONJOB_PERMISSION } as any;
+      }
+      return false;
+    });
+
+    const cronjobController = new CronjobController({
+      messenger: controllerMessenger,
+    });
+
+    await cronjobController.register(MOCK_SNAP_ID);
+
+    const snapInfo: TruncatedSnap = {
+      blocked: false,
+      enabled: true,
+      id: MOCK_SNAP_ID,
+      initialPermissions: {},
+      permissionName: '',
+      version: '',
+    };
+    // eslint-disable-next-line dot-notation
+    cronjobController['_handleEventSnapRemoved'](snapInfo);
+
+    // eslint-disable-next-line dot-notation
+    expect(cronjobController['_snapIds'].size).toBe(0);
+
+    cronjobController.destroy();
+  });
+
+  it('handles SnapUpdated event', async () => {
+    const MOCK_ANOTHER_CRONJOB_PERMISSION = deepClone(MOCK_CRONJOB_PERMISSION);
+    MOCK_ANOTHER_CRONJOB_PERMISSION.caveats[0].value = {
+      jobs: [
+        {
+          expression: '*/15 * * * *',
+          request: {
+            method: 'exampleMethodOne',
+            params: ['p1'],
+          },
+        },
+      ],
+    };
+    const rootMessenger = getRootCronjobControllerMessenger();
+    const controllerMessenger =
+      getRestrictedCronjobControllerMessenger(rootMessenger);
+
+    jest
+      .spyOn(controllerMessenger, 'call')
+      .mockResolvedValueOnce([getTruncatedSnap()])
+      .mockResolvedValueOnce({
+        [SnapEndowments.Cronjob]: MOCK_CRONJOB_PERMISSION,
+      })
+      .mockResolvedValueOnce({
+        [SnapEndowments.Cronjob]: MOCK_ANOTHER_CRONJOB_PERMISSION,
+      });
+
+    const cronjobController = new CronjobController({
+      messenger: controllerMessenger,
+    });
+
+    await cronjobController.register(MOCK_SNAP_ID);
+
+    // eslint-disable-next-line dot-notation
+    expect(cronjobController['_snapIds'].size).toBe(2);
+    // eslint-disable-next-line dot-notation
+    expect(cronjobController['_timers'].size).toBe(2);
+
+    const snapInfo: TruncatedSnap = {
+      blocked: false,
+      enabled: true,
+      id: MOCK_SNAP_ID,
+      initialPermissions: {},
+      permissionName: '',
+      version: '',
+    };
+    // eslint-disable-next-line dot-notation
+    await cronjobController['_handleEventSnapUpdated'](snapInfo);
+
+    // eslint-disable-next-line dot-notation
+    expect(cronjobController['_snapIds'].size).toBe(1);
+    // eslint-disable-next-line dot-notation
+    expect(cronjobController['_timers'].size).toBe(1);
```

Comment:
```suggestion
    await cronjobController._handleEventSnapUpdated(snapInfo);

    expect(cronjobController._snapIds.size).toBe(1);
    expect(cronjobController._timers.size).toBe(1);
```

Reply:
Updated this part with something like this:

```typescript
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-expect-error
    await cronjobController._handleEventSnapUpdated(snapInfo);
    expect(cronjobController.snapIds.size).toBe(1);
    expect(cronjobController.timers.size).toBe(1);
```
_Getters are used to achieve this._